### PR TITLE
add esri-leaflet-related@2.0.0 w/ npm auto-update

### LINKED ISSUE #8861: [Request] Add esri-leaflet-related
**Library name:** esri-leaflet-related
**Git repository url:** https://github.com/jgravois/esri-leaflet-related

### PR DESCRIPTION
**Pull request by nolsherry**
@pvnr0082t, please help me review this PR, thanks!

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/jgravois/esri-leaflet-related
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/esri-leaflet-related
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 3
   - Count of watchers: 4
   - Count of forks: 3
 * NPM download stats (optional):
   - Downloads in the last day: 8
   - Downloads in the last week: 29
   - Downloads in the last month: 73

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [X] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [X] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [X] Has valid tags for each versions (for git auto-update)
 * [X] Auto-update setup
 * [X] Auto-update target/source is valid.
 * [X] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [X] Separate unrelated changes into different commits.
 * [X] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [X] Close corresponding issue in commit message
 * [X] Mention related issue(s), people in commit message, comment.

close #8861, cc @jgravois

### CODE REVIEW

File: ajax/libs/esri-leaflet-related/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  "name": "esri-leaflet-related",
+  "filename": "EsriLeafletRelated.min.js",
+  "description": "A Leaflet plugin that allows users to query related tables.",
+  "version": "2.0.0",
+  "author": "John Gravois <jgravois@esri.com> (http://johngravois.com)",
+  "dependencies": {
+    "leaflet": "^1.0.0-beta.1",
+    "esri-leaflet": "^2.0.0-beta.4"
+  },
+  "jsnext:main": "src/EsriLeafletRelated.js",
```
Comment (pvnr0082t) [inline ajax/libs/esri-leaflet-related/package.json]: @nolsherry Please remove `jsnext:main`, `jspm` and `readmeFilename` field.

**Comment by pvnr0082t** [comment]
Please add `keywords` in package.json. Thanks.

File: ajax/libs/esri-leaflet-related/2.0.0/EsriLeafletRelated.min.js [hunk 2]
```diff
@@ -0,0 +1 @@
+// Error: Unexpected token: keyword (var)
```
Comment (pvnr0082t) [inline ajax/libs/esri-leaflet-related/2.0.0/EsriLeafletRelated.min.js]: Why don't you get minified files under `dist` folder? BTW, you need to check if files are minified successfully.

**Comment by pvnr0082t** [comment]
@nolsherry You need to change the file permission to inexecutable.

File: ajax/libs/esri-leaflet-related/2.0.0/EsriLeafletRelated.min.js (hunk 2 above)
Comment (PeterDaveHello) [inline ajax/libs/esri-leaflet-related/2.0.0/EsriLeafletRelated.min.js]: Nice catch!

**Comment by nolsherry** [comment]
@pvnr0082t, I have removed `jsnext:main`, `jspm` and `readmeFilename` fields.
In additions, I have got minified files under `dist` folder, 
Please help me review this PR, thanks!

**Comment by pvnr0082t** [comment]
@nolsherry You can also grab unminified file from `src` folder.

**Comment by nolsherry** [comment]
@pvnr0082t, I have updated `package.json`, please help me review this PR, thanks!

File: ajax/libs/esri-leaflet-related/package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    {
+      "basePath": "dist",
+      "files": [
+        "**/*"
+      ]
+    },
+    {
+      "basePath": "src",
+      "files": [
+        "**/*"
```
Comment (pvnr0082t) [inline ajax/libs/esri-leaflet-related/package.json]: Let's use `"EsriLeafletRelated*"` here.

**Comment by nolsherry** [comment]
@pvnr0082t, I have updated th `package.json`, please help me review this PR.

**Comment by pvnr0082t** [comment]
LGTM

**Comment by PeterDaveHello** [comment]
In this case, the naming scheme is different between the src and dist files, maybe we can just add the dist file? So that it won't make users confused.

**Comment by pvnr0082t** [comment]
@nolsherry As Peter said, we can just add files under `dist` folder. Please remove `EsriLeafletRelated.js` and modify filemap. Thanks.

**Comment by nolsherry** [comment]
@pvnr0082t, I have modified the filemap, and removed `EsriLeafletRelated.js` file, please help me review this PR, thanks!

**Comment by pvnr0082t** [comment]
@nolsherry You don't fix them. Please check again before you call reviewer! THANKS.

**Comment by nolsherry** [comment]
@pvnr0082t, I am sorry for my mistaking, please help me review this PR again, thanks!

**Comment by pvnr0082t** [comment]
LGTM

**Comment by PeterDaveHello** [comment]
LGTM, @pvnr0082t please merge this.

**Pull request merged**